### PR TITLE
LPS-123794 | LPS-123795 DnD & Resize

### PR DESCRIPTION
**Pull request by diegonvs**
---- 
For helping the reviewer here are the commits:

* LPS-123794 Source formatting

* LPS-123794 Fix missing behaviours

- Adds a FieldDragPreview using the same styles of the field. The DOM was touched because we can't use directly the field since it's loaded lazily.
- Some info need to be stored on a ref on the Row component because React was resetting some values on the Column
- We can't have more than one DragLayer on React tree. Having more than one was causing a bug with multiple DragLayers being rendered everywhere. So, was decided to move the DragLayer to a common parent component(FormNoop).

* LPS-123795 Rewrite field Resize logic from the DOM oriented calculation to a more smooth and simpler code

- We removed `withResizeableColumns`  HOC for disable the legacy implementation
- We needed to replace the [getIndexes](https://github.com/liferay/liferay-portal/blob/b316fa46842332f225925ae23d8e5352dfe82493/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/util/FormSupport.es.js#L366)/[getNestedIndexes](https://github.com/liferay/liferay-portal/blob/b316fa46842332f225925ae23d8e5352dfe82493/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/util/FormSupport.es.js#L376) by a React like solution. We don't need to pollute our elements with too much data-attrs*, so, We are using the `loc` property where it's defined on the ParentFieldContext.
- We are implementing the logic inside the ResizableColumn component inside EditorVariant.Column component
- We needed to do some refactoring for traversing `rowRef` across the EditorVariant(row and column) for calculating the current hovered column.

* LPS-123794 Instead of calling `openToast`, throws the message error on the browser console.

openToast alert omits the stack-trace

* LPS-123794 Adds custom `useDrag` and `useDrop` and creates more item.types

Was created a generic useDrop/useDrag hook for dispatching different actions depending of the type of the Drag.

You may notice We can have two types of Drag:

-  You can drag between fields(useDrag from EditorVariant.Column and Pages)

- You can drag between a sidebar and a builder(useDrag defined on Sidebar components)

This knowledge is necessary for deciding which drag action I need to call. The main idea is to call the action ended with `MOVE` when dropping something initialized with values from the drag between fields and `ADD` when dropping something initialized with values from drag from the sidebar.

Also, We can have three different types of a drag item:

- fieldType:
When the item to be dragged is a field/sidebar field

- dataDefinitionField:
When the item to be dragged came from App Builder CustomObjectList sidebar

In the future it will be replaced by just `fieldset` or `fieldtype` depending on the context

- fieldset:
When the item to be dragged came from a fieldset.

Currently, there are 7 types of Drag, combining the already defined types of field/action:

`DragTypes.DRAG_FIELD_TYPE_ADD`
`DragTypes.DRAG_DATA_DEFINITION_FIELD_MOVE`
`DragTypes.DRAG_FIELD_TYPE_MOVE`
`DragTypes.DRAG_FIELDSET_MOVE`
`DragTypes.DRAG_DATA_DEFINITION_FIELD_ADD`
`DragTypes.DRAG_FIELD_TYPE_ADD`
`DragTypes.DRAG_FIELDSET_ADD`

* LPS-123794 Disables `withMoveableFields` HOC for disabling the legacy drag-and-drop functionallity implemented by metal-drag-drop

⚠️  Warning

This PR is on top of:
https://github.com/liferay-forms/liferay-portal/pull/128

_Disseram que é o novo normal do form-builder._

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js [hunk 1]
```diff
@@ -14,65 +14,141 @@
...
+		const element = containerRef.current.cloneNode(true);
+
+		ref.current.parentElement.style.width = width;
+
+		ref.current.appendChild(element);
+
+		return () => {
+			ref.current.remove();
+		};
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (diegonvs) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js:69]: SF still failing here but I used correctly `eslint-disable-next-line`. I expect when rebuilding entire DXP will solve this on my machine.

Comment (diegonvs) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js]: Solved on e7db112 Thx Matu <3

**Comment by diegonvs** [comment]
Added b751cde28b9c3ac8d3497a8dc59e2b3c23ab9536 and rebased with #128. So, it was the push-force :)

**Comment by diegonvs** [comment]
image

https://www.youtube.com/watch?v=fJb_Jz_ce50

vai filhão

**Comment by diegonvs** [comment]
ci:test:sf

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by sandrodw3** [comment]
🕯

**Comment by p2kmgcl** [comment]
:candle:

**Comment by victorg1991** [comment]
🕯👏🏻

**Comment by diegonvs** [comment]
Conflicts...

image

Force-pushing and rerunning the ci

**Comment by diegonvs** [comment]
Taking a look on https://github.com/liferay-forms/liferay-portal/pull/132#issuecomment-780237163:

`LocalFile.FormRules#DeleteFieldThatBelongsToARule` seems unrelated running locally.

`LocalFile.AppBuilder#CheckRequiredForm` it's running well locally.

`LocalFile.AppBuilder#PropagateChangeTranslatedFieldSetSameObject` it's running well locally.

`LocalFile.AppBuilder#ReusedTranslatedFieldset` seems unrelated.

**Comment by diegonvs** [comment]
ci:test:sf

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
c:test:forms

**Comment by diegonvs** [comment]
ci:test:forms

**Comment by diegonvs** [comment]
ci:test:app-builder

**Comment by diegonvs** [comment]
ci:test:sf

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
When re-running locally these tests are running well. If relevant suite will fail again, I need help

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
ci:test:sf

**Comment by diegonvs** [comment]
ci:test:stop

**Comment by diegonvs** [comment]
ci:stop

**Comment by diegonvs** [comment]
ci:stop

**Comment by diegonvs** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js [hunk 2]
```diff
@@ -112,30 +194,169 @@ export const Column = ({
...
+							field,
+							index,
+							loc: {
+								columnIndex,
+								pageIndex,
+								rowIndex,
+							},
+						})
+					)}
+				</ResizableColumn>
```
Comment (matuzalemsteles) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js:226]: @diegonvs there is a coupling here that ends up that it makes no sense to have the `ResizableColumn` separately. Some things I noticed:

- There is a dependency on `drag` and `drop`
- It depends on `DraggableField` which will not help with the resize part

So maybe we are concentrating logic and responsibilities beyond the scope of Resize for this component, we can move a few things to separate these responsibilities and the name of the component to living up to its responsibility will also be easier to maintain.

Maybe just moving `DraggableField` (Read my comment on that component too) here already solves practically the entire liability problem. Consider the following:

```suggestion
				<ResizableColumn
					currentLoc={{columnIndex, pageIndex, rowIndex}}
					disabled={!isFieldSelected || !editable}
					instanceId={firstField.instanceId}
					onResizing={(resizing) => setResizing(resizing)}
					resizeInfoRef={resizeInfoRef}
					rowRef={rowRef}
				>
					<div
						className={classNames('ddm-drag', {
							'py-0': isFieldSetOrGroup
						})}
						ref={(node) => {
							if (allowNestedFields && !rootParentField.ddmStructureId) {
								drag(drop(node));
							}
							else if (!hasFieldSet(parentField)) {
								drag(node);
							}

							resizeRef.current = node;
						}}
					>
						{column.fields.map((field, index) =>
							children({
								field,
								index,
								loc: {
									columnIndex,
									pageIndex,
									rowIndex,
								},
							})
						)}
					</div>
				</ResizableColumn>
```

With that we also removed some dependencies of some properties that `ResizableColumn` had due to the current implementation of this PR. Some things for notes:

- `disabled` is a more friendly API than passing `isFieldSelected` and `editable` to the component which in the end it decides whether to disable the resize handle
- `currentLoc` the final idea is to provide the loc of the current field, so instead of passing each index of the loc as a property to the component, we can only pass the assembled loc since it is the final objective of the component and we managed to decrease the amount of APIs.
- `drag`, `drop`, `allowNestedFields`, `rootParentField`, `parentField`, `isFieldSetOrGroup` and `ref=resizeRef` have been removed since we no longer have the `drag` and `drop` coupling inside the resize.

One thing to keep in mind is, try to create APIs that can help you understand the final purpose of the component, an example of this is passing `disabled` instead offer `isFieldSelected` and `editable`. The big goal is to provide a component that can cover a responsibility well or allow it to be reused.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js [hunk 3]
```diff
@@ -14,65 +14,144 @@
...
+import {EVENT_TYPES} from '../../actions/eventTypes.es';
 import {DND_ORIGIN_TYPE, useDrop} from '../../hooks/useDrop.es';
+import {useForm} from '../../hooks/useForm.es';
 import {hasFieldSet} from '../../util/fields.es';
 import {Actions, ActionsControls, useActions} from '../Actions.es';
 import {ParentFieldContext} from '../Field/ParentFieldContext.es';
 import {Placeholder} from '../Placeholder.es';
 import * as DefaultVariant from './DefaultVariant.es';
 
+const DIRECTIONS = {
```
Comment (matuzalemsteles) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js:38]:
```suggestion
const RESIZE_DIRECTIONS = {
```

File: modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/drag-and-drop/getDropHandler.es.js [hunk 4]
```diff
@@ -18,9 +18,9 @@ import {
...
 import {
-	DRAG_DATA_DEFINITION_FIELD,
-	DRAG_FIELDSET,
-	DRAG_FIELD_TYPE,
+	DRAG_DATA_DEFINITION_FIELD_ADD,
+	DRAG_FIELDSET_ADD,
+	DRAG_FIELD_TYPE_ADD,
 } from './dragTypes.es';
 
 export const getDropHandler = ({dataDefinition, dataLayoutBuilder}) => {
```
Comment (matuzalemsteles) [inline modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/drag-and-drop/getDropHandler.es.js:26]: @diegonvs I think Aline and Milton probably forgot to remove this file. Since we have all the implementations in `useDrop` this file no longer makes sense and is no longer being used. Please can remove it in your PR. It is also necessary to remove this `useEffect` https://github.com/liferay-forms/liferay-portal/blob/bd1ec4db9005b677f34046bff05ad71e75b411c9/modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/data-layout-builder/DataLayoutBuilderContextProvider.es.js#L121-L128

cc @miltonmc @alinedoleron

File: modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/pages/table-view/DropZone.es.js [hunk 5]
```diff
@@ -46,7 +46,7 @@ const generateItems = (columns, rows = 10) => {
 
 const DropZone = ({fields, onAddFieldName, onRemoveFieldName}) => {
 	const [{canDrop, overTarget}, drop] = useDrop({
-		accept: DragTypes.DRAG_FIELD_TYPE,
```
Comment (matuzalemsteles) [inline modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/pages/table-view/DropZone.es.js:49]: @diegonvs can you check if we should change here too? I think this is for another screen that is not related to Form Builder, just to check, try to create a Form View and Table View.

File: modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/pages/form-view/CustomObjectFieldsList.es.js [hunk 6]
```diff
@@ -263,6 +263,21 @@ const CustomObjectFieldsList = ({keywords}) => {
...
+		const settingsContext = dataLayoutBuilder.getDDMFormFieldSettingsContext(
+			dataDefinitionField
+		);
+
+		return {
+			...dataDefinitionField,
+			settingsContext,
+		};
+	};
+
```
Comment (matuzalemsteles) [inline modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/pages/form-view/CustomObjectFieldsList.es.js:280]: You probably missed this during the rebase, but this is no longer necessary with the changes of Aline and Milton, it is now called directly in the DataConverter.

File: modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/field-types/FieldTypeList.es.js [hunk 7]
```diff
@@ -74,6 +75,7 @@ const FieldTypeList = ({
 				<FieldTypeWrapper
 					dataDefinition={dataDefinition}
 					deleteLabel={deleteLabel}
+					dragType={DRAG_DATA_DEFINITION_FIELD_ADD}
```
Comment (matuzalemsteles) [inline modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/field-types/FieldTypeList.es.js:78]: You do not need to define `dragType` here, because it is already being defined in `CustomObjectFieldsList`. https://github.com/liferay-forms/liferay-portal/blob/bd1ec4db9005b677f34046bff05ad71e75b411c9/modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/pages/form-view/CustomObjectFieldsList.es.js#L126-L127 

Also remembering that this component is reused by the Sidebar, but its `dragType` has to be `DRAG_FIELD_TYPE_ADD` if you do this here, it will be breaking the Sidebar.

File: modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/drag-and-drop/dragTypes.es.js [hunk 8]
```diff
@@ -12,6 +12,10 @@
  * details.
  */
 
-export const DRAG_DATA_DEFINITION_FIELD = 'dataDefinitionField';
-export const DRAG_FIELD_TYPE = 'fieldType';
-export const DRAG_FIELDSET = 'fieldset';
+export const DRAG_DATA_DEFINITION_FIELD_MOVE = 'dataDefinitionField:move';
+export const DRAG_FIELD_TYPE_MOVE = 'fieldType:move';
+export const DRAG_FIELDSET_MOVE = 'fieldset:move';
```
Comment (matuzalemsteles) [inline modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/drag-and-drop/dragTypes.es.js:17]: There must be no `DRAG_DATA_DEFINITION_FIELD_MOVE` and `DRAG_FIELDSET_MOVE` because they are all `DRAG_FIELD_TYPE_MOVE`, there is no difference when moving.

**Review by matuzalemsteles — CHANGES_REQUESTED** [review]
@diegonvs I have added a lot of comments regarding some things that you may have missed that may impact the application in some places and we may have some regressions, please check each detail carefully and at the end, we can review what has been done.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/hooks/useDrop.es.js [hunk 9]
```diff
@@ -139,6 +163,22 @@ export const useDrop = ({
...
+							targetIndexes: {
+								columnIndex,
+								pageIndex,
+								rowIndex,
+							},
+							targetParentFieldName: parentField?.fieldName,
+						},
+						type: EVENT_TYPES.FIELD_MOVED,
+					});
+					break;
```
[the remark this answers is not in the document]
Comment (diegonvs) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/hooks/useDrop.es.js:181]: Makes sense. Removing `DRAG_FIELDSET_MOVE` dragType in favor to use just `DRAG_FIELD_TYPE_MOVE ` ;)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/hooks/useDrop.es.js [hunk 10]
```diff
@@ -40,10 +40,18 @@ export const useDrop = ({
 	const dispatch = useForm();
 
 	const [{canDrop, overTarget}, drop] = useDndDrop({
-		accept: ['dataDefinitionField', 'fieldType', 'fieldset'],
+		accept: [
+			DragTypes.DRAG_FIELD_TYPE_ADD,
+			DragTypes.DRAG_DATA_DEFINITION_FIELD_MOVE,
```
[the remark this answers is not in the document]
Comment (diegonvs) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/hooks/useDrop.es.js:45]: Hawkeyes! Thanks for point up this!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js [hunk 11]
```diff
@@ -14,65 +14,144 @@
...
 		fieldName: column.fields[0]?.fieldName,
 		origin: DND_ORIGIN_TYPE.FIELD,
 		pageIndex,
 		parentField,
 		rowIndex,
 	});
 
+	const dragType = isFieldSet
+		? DragTypes.DRAG_FIELDSET_MOVE
+		: DragTypes.DRAG_FIELD_TYPE_MOVE;
```
[the remark this answers is not in the document]
Comment (diegonvs) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js:123]: Solved here:
https://github.com/liferay-forms/liferay-portal/pull/132#discussion_r581438625

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js [hunk 12]
```diff
@@ -112,30 +194,169 @@ export const Column = ({
...
+					)}
+				</ResizableColumn>
+			</DefaultVariant.Column>
+		</ActionsControls>
+	);
+};
+
+Column.displayName = 'EditorVariant.Column';
+
+const ResizableColumn = forwardRef(
```
[the remark this answers is not in the document]
Comment (diegonvs) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js:234]: Makes sense! Moved!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js [hunk 13]
```diff
@@ -112,30 +194,169 @@ export const Column = ({
...
+			resizeInfoRef.current = {
+				...resizeInfoRef.current,
+				direction,
+				instanceId,
+			};
+
+			onResizing(true);
+		};
+
+		const handleMouseMove = (event) => {
```
[the remark this answers is not in the document]
Comment (diegonvs) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js:273]: > useEventListener is removing the event and adding it again with each rendering?

Yes, it was. When hoisting the function using useCallback it prevented this weird behaviour ;)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js [hunk 14]
```diff
@@ -112,30 +194,169 @@ export const Column = ({
...
+
+		useEventListener('mousemove', handleMouseMove, false, document.body);
+
+		useEventListener(
+			'mouseup',
+			() => {
+				onResizing(false);
+
+				resizeInfoRef.current = null;
+			},
```
[the remark this answers is not in the document]
Comment (diegonvs) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js:319]: https://github.com/liferay-forms/liferay-portal/pull/132#discussion_r582008088

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js [hunk 15]
```diff
@@ -14,65 +14,144 @@
...
 import * as DefaultVariant from './DefaultVariant.es';
 
+const DIRECTIONS = {
+	LEFT: 'left',
+	RIGHT: 'right',
+};
+
+const MAX_COLUMNS = 12;
+
+const FieldDragPreview = ({containerRef}) => {
```
[the remark this answers is not in the document]
Comment (diegonvs) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js:45]: Moving...

jason

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js [hunk 16]
```diff
@@ -14,65 +14,144 @@
...
+			<div ref={ref} />
+		</DraggableField>
+	);
+};
+
+const DraggableField = forwardRef(({children, className}, ref) => (
+	<div className={classNames('ddm-drag', className)} ref={ref}>
+		{children}
+	</div>
+));
```
[the remark this answers is not in the document]
Comment (diegonvs) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js:83]: Makes sense, just placing more one point to don't use this component is because We're forwarding the reference.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js (hunk 2 above)
Comment (diegonvs) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js:226]: > One thing to keep in mind is, try to create APIs that can help you understand the final purpose of the component, an example of this is passing disabled instead offer isFieldSelected and editable. The big goal is to provide a component that can cover a responsibility well or allow it to be reused.


💖

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js (hunk 3 above)
Comment (diegonvs) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js:38]: Done!

File: modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/drag-and-drop/getDropHandler.es.js (hunk 4 above)
Comment (diegonvs) [reply modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/drag-and-drop/getDropHandler.es.js:26]: Removed! 

@interaminense @kevenleone

File: modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/pages/table-view/DropZone.es.js (hunk 5 above)
Comment (diegonvs) [reply modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/pages/table-view/DropZone.es.js:49]: Nope, I'll not change it. Table View is currently using EditorVariant's useDrop and the useDrag from the FieldType component. When hard-coding `fieldType` it won't work

File: modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/pages/form-view/CustomObjectFieldsList.es.js (hunk 6 above)
Comment (diegonvs) [reply modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/pages/form-view/CustomObjectFieldsList.es.js:280]: Yes, it's unused. Removing!

File: modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/field-types/FieldTypeList.es.js (hunk 7 above)
Comment (diegonvs) [reply modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/field-types/FieldTypeList.es.js:78]: > You do not need to define dragType here, because it is already being defined in CustomObjectFieldsList.

Yes, makes sense. I remember to add this dragType here a month ago because when dragging dataDefinitionFields weren't working but without this code it also works properly ;)

File: modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/drag-and-drop/dragTypes.es.js (hunk 8 above)
Comment (diegonvs) [reply modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/drag-and-drop/dragTypes.es.js:17]: It was previously removed! Thanks for reviewing Matu!

**Comment by diegonvs** [comment]
ci:test:sf

**Comment by diegonvs** [comment]
ci:test:sf

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
ci:reevaluate:1207401_9302

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by natocesarrego** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by natocesarrego** [comment]
I just downloaded the pull to analyze and fix the App Builder reported failure.

**Comment by natocesarrego** [comment]
ci:test:sf

**Comment by natocesarrego** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
ci:test:app-builder

**Comment by diegonvs** [comment]
ci:forward
